lib/mongo_ecto/connection.ex

```diff
@@ -124,22 +124,29 @@ defmodule Mongo.Ecto.Connection do
       fun.()
     rescue
       e in Mongo.Error ->
-        stacktrace = System.stacktrace
         case e do
           %Mongo.Error{code: 11000, message: msg} ->
             {:invalid, constraint(msg)}
           other ->
+            stacktrace = System.stacktrace
```

Comment:
Other code can change the stacktrace which will cause very confusing error messages. Because of this you should always capture the stacktrace as early as possible.